targets/csharp/source/source/PlayFabGameServerInternal.cs

```diff
@@ -0,0 +1,310 @@
+        private bool _shouldWriteDebugLogs;
+
+        public ManualResetEvent TransitionToActiveEvent { get; set; }
+        public IDictionary<string, string> ConfigMap { get; private set; }
+        public ILogger Logger { get; private set; }
+        public GameState State
+        {
+            get { return _state; }
+            set
+            {
+                if (_state != value)
+                {
+                    _state = value;
+                    _signalHeartbeatEvent.Set();
+                }
+            }
+        }
+
+        public IList<GameServerConnectedPlayer> ConnectedPlayers { get; set; }
+
+        public IList<string> InitialPlayers { get; set; }
+
+        // Keep an instance of the callbacks around so they don't get garbage collected
+        public Action ShutdownCallback { get; set; }
+        public Func<bool> HealthCallback { get; set; }
+        public Action<DateTimeOffset> MaintenanceCallback { get; set; }
+
+
+        public GameServerInternalSdk(string configFileName = null)
+        {
+            _overrideConfigFileName = configFileName;
+            ConnectedPlayers = new List<GameServerConnectedPlayer>();
+            InitialPlayers = new List<string>();
+            TransitionToActiveEvent = new ManualResetEvent(false);
+        }
+
+        public Task StartAsync(bool shouldWriteDebugLogs = false)
+        {
+            // If we already initialized everything, no need to do it again
+            if (_heartbeatTask != null)
+            {
+                return Task.CompletedTask;
+            }
+
+            _shouldWriteDebugLogs = shouldWriteDebugLogs;
+
+            _configuration = _configuration ?? GetConfiguration();
+            ConfigMap = ConfigMap ?? CreateConfigMap(_configuration);
+            Logger = LoggerFactory.CreateInstance(ConfigMap[PlayFabGameServerSDK.LogFolderKey]);
+            
+            if (_configuration.ShouldLog())
+            {
+                Logger.Start();
+            }
+
+            string gsmsBaseUrl = ConfigMap[PlayFabGameServerSDK.HeartbeatEndpointKey];
+            string instanceId = ConfigMap[PlayFabGameServerSDK.ServerIdKey];
+
+            Logger.Log($"VM Agent Endpoint: {gsmsBaseUrl}");
+            Logger.Log($"Instance Id: {instanceId}");
+
+            _webClient = GameServerHttpClientFactory.CreateInstance($"http://{gsmsBaseUrl}/v1/sessionHosts/{instanceId}");
+
+            _heartbeatRunning = true;
+            _signalHeartbeatEvent.Reset();
+            TransitionToActiveEvent.Reset();
+
+            _heartbeatTask = Task.Run(HeartbeatAsync);
+            return Task.CompletedTask;
+        }
+
+        private GameServerConfiguration GetConfiguration()
+        {
+            string fileName = !string.IsNullOrWhiteSpace(_overrideConfigFileName) ? _overrideConfigFileName
+                                : Environment.GetEnvironmentVariable(PlayFabGameServerSDK.GsdkConfigFileEnvVarKey);
```

Comment:
Weird tabbing here.
In VS, Please press "Ctrl + K + D" to standardize all spacing and tabbing in all files.
(It doesn't always fix every line, so double check this one, and manually fix it if necessary.)

Reply:
I'll just go back to using if statements since it's too long with the ternary operator, sorry Andrew XD

---
In reply to: [256124424](https://github.com/PlayFab/SDKGenerator/pull/398#discussion_r256124424) [](ancestors = 256124424)